Fix type annotations in pandas.core.indexes.period

Part of #25882 

Current errors are:

    pandas\core\indexes\period.py:171: error: Incompatible types in assignment (expression has type "PeriodArray", base class "DatetimeIndexOpsMixin" defined the type as "None")

--- a/mypy.ini
+++ b/mypy.ini
@@ -7,6 +7,3 @@ ignore_errors=True
 
 [mypy-pandas.core.indexes.datetimelike]
 ignore_errors=True
-
-[mypy-pandas.core.indexes.period]
-ignore_errors=True

--- a/pandas/core/indexes/period.py
+++ b/pandas/core/indexes/period.py
@@ -80,7 +80,7 @@ class PeriodIndex(DatetimeIndexOpsMixin, Int64Index, PeriodDelegateMixin):
 
     Parameters
     ----------
-    data : array-like (1-dimensional) - PeriodArray, optional
+    data : array-like (1d integer np.ndarray or PeriodArray), optional
         Optional period-like data to construct index with
     copy : bool
         Make a copy of input ndarray

--- a/pandas/core/indexes/period.py
+++ b/pandas/core/indexes/period.py
@@ -80,7 +80,7 @@ class PeriodIndex(DatetimeIndexOpsMixin, Int64Index, PeriodDelegateMixin):
 
     Parameters
     ----------
-    data : array-like (1-dimensional), optional
+    data : array-like (1-dimensional) - PeriodArray, optional
         Optional period-like data to construct index with
     copy : bool
         Make a copy of input ndarray
@@ -168,7 +168,7 @@ class PeriodIndex(DatetimeIndexOpsMixin, Int64Index, PeriodDelegateMixin):
     _is_numeric_dtype = False
     _infer_as_myclass = True
 
-    _data = None  # type: PeriodArray
+    _data = None
 
     _engine_type = libindex.PeriodEngine
 

